prow/github/client.go

```diff
@@ -3593,6 +3593,8 @@ func (c *client) CreateTeam(org string, team Team) (*Team, error) {
 	if c.fake {
 		return nil, nil
 	} else if c.dry {
+		// When in dry mode we need a believable slug to call corresponding methods for this team
+		team.Slug = strings.ToLower(strings.ReplaceAll(team.Name, " ", "-"))
```

Comment:
It's not really required since it won't be used to actually call a gh api. The only important thing about this slug is that it is unique for the org, which this will be as long as the team names are unique (and that would be an error if they weren't). Even the spaces didn't need to be escaped, but that makes it look like a _real_ slug. I don't think we need to handle every edge case and recreate the slug creation logic here, but I am willing to escape more chars and cap it if you think it's important.